pkg/cli/admin/release/mirror.go

```diff
@@ -442,18 +450,67 @@ func (o *MirrorOptions) Run() error {
 	} else {
 		fmt.Fprintf(o.Out, "\nSuccess\nUpdate image:  %s\nMirrored to: %s\n", to, o.To)
 	}
+
+	printImageContentInstructions(o.Out, o.From, o.To, repositories)
 	return nil
 }
 
-func sourceImageRef(is *imagev1.ImageStream, name string) (string, bool) {
-	for _, tag := range is.Spec.Tags {
-		if tag.Name != name {
-			continue
-		}
-		if tag.From == nil || tag.From.Kind != "DockerImage" {
-			return "", false
-		}
-		return tag.From.Name, true
+// printImageContentInstructions provides exapmles to the user for using the new repository mirror
+// https://github.com/openshift/installer/blob/master/docs/dev/alternative_release_image_sources.md
+func printImageContentInstructions(out io.Writer, from, to string, repositories map[string]struct{}) error {
+	type imageContentSource struct {
+		Source  string   `json:"source"`
+		Mirrors []string `json:"mirrors"`
+	}
+	type installConfigSubsection struct {
+		ImageContentSources []imageContentSource `json:"imageContentSources"`
+	}
+
+	type icspSubsection struct {
+		RepositoryDigestMirrors []imageContentSource `json:"repositoryDigestMirrors"`
 	}
-	return "", false
+
+	var sources []imageContentSource
+
+	mirrorRef, err := imagereference.Parse(to)
+	if err != nil {
+		return fmt.Errorf("Unable to parse image reference '%s': %v", to, err)
+	}
+	mirrorRepo := mirrorRef.AsRepository().String()
+
+	sourceRef, err := imagereference.Parse(from)
+	if err != nil {
+		return fmt.Errorf("Unable to parse image reference '%s': %v", from, err)
+	}
+	sourceRepo := sourceRef.AsRepository().String()
+	repositories[sourceRepo] = struct{}{}
+
+	for repository := range repositories {
+		sources = append(sources, imageContentSource{
+			Source:  repository,
+			Mirrors: []string{mirrorRepo},
+		})
+	}
+
+	// Create and display install-config.yaml example
+	imageContentSources := installConfigSubsection{
+		ImageContentSources: sources}
+	installConfigExample, err := yaml.Marshal(imageContentSources)
+	if err != nil {
+		return fmt.Errorf("Unable to marshal imageContentSources yaml: %v", err)
+	}
+	fmt.Fprintf(out, "\nTo use the new mirrored repository to install, add the following section to the install-config.yaml:\n")
+	fmt.Fprintf(out, string(installConfigExample))
+
+	// Create and display ImageContentSourcePolicy example
+	repositoryDigestMirrors := icspSubsection{
+		RepositoryDigestMirrors: sources}
+	icspExample, err := yaml.Marshal(repositoryDigestMirrors)
+	if err != nil {
+		return fmt.Errorf("Unable to marshal repositoryDigestMirrors yaml: %v", err)
+	}
+	fmt.Fprintf(out, "\nTo use the new mirrored repository for upgrades, add the following section to an ImageContentSourcePolicy:\n")
```

Comment:
Nitpick: Should this mention the `Spec` field of `ImageContentSourcePolicy`?

Reply:
I think the general goal for the output here is going to be provide information that helps users to take direct action.

So the first part if going to be, how to install using that mirror.
and 
the first part provides sub-section of the install-config because otherwise the whole object is difficult to provide.

The second question is, how to upgrade to that mirror.
and I think since `github.com/openshift/api` is already vendored.. providing an **complete** object that can be pushed to the cluster to enable the upgrade has it's merit.

cc @smarterclayton

Reply:
We can always take feedback on this, as long as it's not too hard to do each step.